compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -229,6 +229,48 @@ static Value distributeVectors(ImplicitLocOpBuilder &b, Value funcH,
   return funcH;
 }
 
+/// Various handles produced by reduction splitting.
+struct ReductionSplitResult {
+  /// Handle to the leading elementwise operation, may be null if no such
+  /// operation is present.
+  Value leadingEltwiseH;
+  /// Handle to the fill operation feeding the init of a higher-rank
+  /// more-parallel reduction.
+  Value splitFillH;
+  /// Handle to the higher-rank more-parallel reduction.
+  Value splitLinalgH;
+  /// Handle to the final reduction.
+  Value combinerH;
+  /// Handle to the original fill operation, may be null if the operation was
+  /// not re-matched.
+  Value originalFillH;
+};
+
+/// Builds transform IR requesting to bubble up the "expand_shape" operation
+/// produced as parent of reduction splitting if necessary for fusion of the
+/// leading elementwise operation.
+static ReductionSplitResult buildExpansionBubbleUp(
```

Comment:
Can we split this in 2? 
There is a generic builder of expansionBubbleOp that is independent of the ops involved.

Reply:
Sure, but not in this PR.